### PR TITLE
v1.0: New error type :`system`

### LINKED ISSUE #2082: v1.0: Improve error handler
## Reference
- Issue on core: https://github.com/meilisearch/meilisearch/issues/3095
- Spec: https://github.com/meilisearch/specifications/pull/212
- Notion: https://www.notion.so/meilisearch/Enhance-DX-Error-Handling-Focus-017a422e30bd4febb52841a463191c0b
- SME:

### LINKED ISSUE #2087: v1.0: new error type
_Check issue on core for updates_

Add a new error type for configuration called `system`.

## Reference
- Issue on core: https://github.com/meilisearch/meilisearch/issues/3271
- Spec:
- SME: @irevoire

### PR DESCRIPTION
**Pull request by maryamsulemani97**
closes #2087

### CODE REVIEW

**Review by irevoire — APPROVED** [review]
purfect thanks! 🐈

File: reference/errors/overview.md [hunk 1]
```diff
@@ -18,11 +18,12 @@ Meilisearch uses the following standard HTTP codes for a successful or failed AP
...
+| :-------------------- | :---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| **`invalid_request`** | This is due to an error in the user input. It is accompanied by the HTTP code `4xx`                                                                                                       |
+| **`internal`**        | This is due to machine or configuration constraints. It is accompanied by the HTTP code `5xx`                                                                                             |
+| **`auth`**            | This type of error is related to authentication and authorization. It is accompanied by the HTTP code `4xx`                                                                               |
+| **`system`**          | This indicates your system has reached or exceeded its limit for disk size, index size, etc., or the database doesn't have read or write access. It is accompanied by the HTTP code `5xx` |
```
Comment (guimachiavelli) [inline reference/errors/overview.md:26]: Can we rewrite this without using "etc"? Either we list all causes or state a general category ("This indicates your system has reached or exceeded its disk space usage limits)—"etc" is sort of in-between and leaves too much open for interpretation.

**Review by guimachiavelli — CHANGES_REQUESTED** [review]

**Review by guimachiavelli — APPROVED** [review]
🥡

**Comment by guimachiavelli** [comment]
bors merge

**Pull request merged**